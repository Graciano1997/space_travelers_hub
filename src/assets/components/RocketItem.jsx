import PropTypes from 'prop-types';

import style from '../style/Rocket.module.css';

const RocketItem = ({
  id, name, description, imageRocket,
}) => (
  <div className={style.rocketItem}>
    <div className={style.rocketImageContainer}><img src={imageRocket[0]} alt="rocket" /></div>
    <div className={style.rocketBody}>
      <h3>{name}</h3>
      <p>{description}</p>
      <p>{id}</p>
      <button type="button" aria-label="reserve / calncel  rocket">Reserve Rocket</button>
    </div>
  </div>
);

RocketItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  // type: PropTypes.string.isRequired,
  imageRocket: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default RocketItem;
