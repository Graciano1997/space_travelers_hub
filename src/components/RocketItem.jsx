import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import Badge from 'react-bootstrap/Badge';
import { Button } from 'react-bootstrap';
import { bookingRockets } from '../redux/rockets/rocketsSlice';
import style from '../assets/style/Rocket.module.css';

function RocketItem({
  id, name, description, imageRocket, reserved,
}) {
  const dispatch = useDispatch();
  const handlerReserve = () => {
    dispatch(bookingRockets(id));
  };
  const info = (reserved) ? 'Cancel Reservation' : 'Reserve Rocket';
  return (
    <div className={style.rocketItem}>
      <div className={style.rocketImageContainer}><img src={imageRocket[0]} alt="rocket" /></div>
      <div className={style.rocketBody}>
        <h3>{name}</h3>
        <p>
          {reserved && (
          <Badge bg="primary" className={style.badge}>
            Reserved
          </Badge>
          )}
          {description}
        </p>
        <Button as="button" variant={(reserved) ? 'outline-secondary' : 'primary'} onClick={handlerReserve} className={style.btnReserve}>
          {info}
        </Button>
      </div>
    </div>
  );
}
RocketItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  reserved: (PropTypes.bool),
  imageRocket: PropTypes.arrayOf(PropTypes.string).isRequired,
};

RocketItem.defaultProps = {
  reserved: undefined,
};

export default RocketItem;
