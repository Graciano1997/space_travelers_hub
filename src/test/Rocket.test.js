import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';
import RocketItem from '../components/RocketItem';

function MockingRockets() {
  return (
    <Provider store={store}>
      <RocketItem
        key={1}
        id={1}
        name="falcon"
        description="good"
        imageRocket={['dfasdf']}
        type="any"
        reserved={false}
      />
    </Provider>
  );
}

describe('testing the Rockets', () => {
  test('the item ', () => {
    const { tree } = render(<MockingRockets />);
    expect(tree).toMatchSnapshot();
  });

  test('the item ', () => {
    render(<MockingRockets />);
    const el = screen.getByText(/falcon/i);
    expect(el).toBeInTheDocument();
  });
});
