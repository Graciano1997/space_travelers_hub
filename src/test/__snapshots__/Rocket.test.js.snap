// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Rockets the all  1`] = `undefined`;

exports[`testing the Rockets the amount of  Rockets and implicitly the getRockets()  1`] = `undefined`;

exports[`testing the Rockets the item  1`] = `undefined`;
